pkg/apis/core/validation/validation.go

```diff
@@ -6000,6 +6001,80 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
+	containersWithHostProcessSetFalse := []string{}
+	containersWhichInheritHostProcessTrue := []string{}
+
+	var podHostProcess *bool
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.WindowsOptions != nil && podSpec.SecurityContext.WindowsOptions.HostProcess != nil {
+		podHostProcess = podSpec.SecurityContext.WindowsOptions.HostProcess
+	}
+
+	hostNetwork := false
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.HostNetwork == true {
+		hostNetwork = true
+	}
+
+	// Note: Pod-wide WindowsSecurityContextOptions are inherited by containers that do not explicitly set them.
+	for _, c := range append(podSpec.Containers, podSpec.InitContainers...) {
+		containerCount++
+		if c.SecurityContext != nil && c.SecurityContext.WindowsOptions != nil && c.SecurityContext.WindowsOptions.HostProcess != nil {
+			if *c.SecurityContext.WindowsOptions.HostProcess {
+				containersWithHostProcessSetTrue = append(containersWithHostProcessSetTrue, c.Name)
+			} else if !*c.SecurityContext.WindowsOptions.HostProcess {
+				containersWithHostProcessSetFalse = append(containersWithHostProcessSetFalse, c.Name)
+			}
+		} else if podHostProcess != nil && *podHostProcess {
+			containersWhichInheritHostProcessTrue = append(containersWhichInheritHostProcessTrue, c.Name)
+		}
+	}
+
+	hostProcessContainerCount := len(containersWithHostProcessSetTrue) + len(containersWhichInheritHostProcessTrue)
+	if hostProcessContainerCount > 0 {
+		// Fail Pod validation if feature is not enabled instead of dropping fields based on PRR reivew.
+		if !utilfeature.DefaultFeatureGate.Enabled(features.WindowsHostProcessContainers) {
+			errMsg := "Pod contains Windows HostProcess containers but feature gate 'WindowsHostProcessContainers' is not enabled"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		// If Pod's WindowsOptions.HostProcess is set to false, no container may set WindowsOptions.HostProcess to true.
+		if podHostProcess != nil && !*podHostProcess && len(containersWithHostProcessSetTrue) > 0 {
+			errMsg := fmt.Sprintf("Pod's WindowsOptions.HostProcess is set to false but the following containers override WindowsOptions.HostProcess to true: %v", containersWithHostProcessSetTrue)
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		// At present, if a Windows Pods contains any HostProcess containers than all containers must be
+		// HostProcess containers (explicitly set or inherited).
+		if hostProcessContainerCount != containerCount {
+			errMsg := "If Pod contains any HostProcess containers then all containers must be HostProcess containers"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
+		}
+
+		// At present Windows Pods which contain HostProcess containers must also set HostNetwork.
+		if hostNetwork != true {
+			errMsg := "PodSecurityContext.HostNetwork must be set if Pod contains any HostProcess containers"
+			allErrs = append(allErrs, field.Forbidden(fieldPath, errMsg))
```

Comment:
In almost all of the failure cases here it is a combination of fields that leads to the error, not any specific field.
I couldn't come up with a consistent way to log errors for most of these cases so thought it would be better to set the pod spec as the field for most of these cases.

Do you have any suggestions on which field I should add the use set as triggering the error?
I didn't see any similar examples to follow here but am open to suggestions.

Reply:
I updated the section a bit and use fields in the error in some of the cases now. PTAL.